app/assets/javascripts/discourse/app/controllers/discovery/list.js

```diff
@@ -207,7 +207,7 @@ export default class DiscoveryListController extends Controller {
     this.bulkSelectHelper.dismissRead(operationType, {
       categoryId: this.model.category?.id,
       tagName: this.model.tag?.id,
-      includeSubcategories: this.model.noSubcategories,
+      includeSubcategories: !this.model.noSubcategories,
```

Comment:
It feels a bit counter-intuitive to use this, but I think it's the correct thing to do. When `noSubcategories=true`, topics from subcategories aren't listed, so we shouldn't dismiss them. Vice-versa, when it is false, topics from subcategories are shown, so they should be included in the list of topics to dismiss.